File: Going Dutch/MCSharedBill.m
//
//  MCSharedBill.m
//  We all pay
//
//  Created by Mark Cornelisse on 23/07/14.
//  Copyright (c) 2014 Mark Cornelisse. All rights reserved.
//

#import "MCSharedBill.h"
#import "MCCurrency.h"
#import "MCPayment.h"
#import "MCPerson.h"
#import "MCCurrency+CoreDataProperties.h"
#import "CurrencyConverter/CurrencyConverter.h"

#import "We_all_pay-Swift.h"

@implementation MCSharedBill

#pragma mark - NSManagedObject

- (void)awakeFromInsert {
    [super awakeFromInsert];
    
    [self setPrimitiveValue:NSUUID.UUID.UUIDString forKey:@"uniqueBillId"];
    [self setPrimitiveValue:@NO forKey:@"hasTheMailBeenSent"];
    NSDate *now = NSDate.date;
    [self setPrimitiveValue:now forKey:@"dateCreated"];
    [self setPrimitiveValue:now forKey:@"dateModified"];
    MCCurrencyModel *currencyModel = [[MCCurrencyModel alloc] initWithManagedObjectContext:self.managedObjectContext andWithCurrencyController:[[CurrencyController alloc] init]];
    NSError *error;
    MCCurrency *mainCurrency = [currencyModel generateCurrencyFromSelectedLocaleWithError:&error];
    NSParameterAssert(error == nil);
    [self setPrimitiveValue:mainCurrency forKey:@"mainCurrency"];
}

#pragma mark - NSObject

@end

File: Going Dutch/MCSharedBill.swift
//
//  MCSharedBill.swift
//  We all pay
//
//  Created by Mark Cornelisse on 26/08/2025.
//  Copyright © 2025 Mark Cornelisse. All rights reserved.
//

import Foundation
import CoreData

import CurrencyConverter

@objc(MCSharedBill) public final class MCSharedBill: NSManagedObject {
    
    // MARK: NSManagedObject
    
    public override func awakeFromInsert() {
        super.awakeFromInsert()
        
        self.setPrimitiveValue(UUID().uuidString, forKey: "uniqueBillId")
        self.setPrimitiveValue(NSNumber(value: false), forKey: "hasTheMailBeenSent")
        let now = Date()
        self.setPrimitiveValue(now, forKey: "dateCreated")
        self.setPrimitiveValue(now, forKey: "dateModified")
        let currencyController = CurrencyController()
        let currencyModel = CurrencyModel(managedObjectContext: self.managedObjectContext!, currencyController: currencyController)
        let mainCurrency = try! currencyModel.generateCurrencyFromSelectedLocale()
        self.setPrimitiveValue(mainCurrency, forKey: "mainCurrency")
    }
    
    // MARK: NSObject
    
}
